### PR TITLE
Cache the RTC value

### PR DESCRIPTION
**Pull request by rcunrau**
Reading the RTC is not for the faint of heart, and also not necessary most of the time. Rather than doing 6(!) I2C transactions _every_ time we need the Unix timestamp, it is far more efficient (and re-entrant) to just read a cached time_t. The cached value is updated by the RTC interrupt once per second.

### CODE REVIEW

File: ex2_hal/athena/equipment_handler/include/rtcmk.h [hunk 1]
```diff
@@ -10,335 +10,35 @@
...
 
 /*******************************************************************************
  *****************************   PROTOTYPES   **********************************
  ******************************************************************************/
 
+/* unix_timestamp is set when RTCMK_SetUnix is called, and then updated every
+ * second by an RTC interrupt. RTCMK_Unix_Now() is the preferred way of getting
+ * the current unix time.
+ */
+static inline time_t RTCMK_Unix_Now() { return unix_timestamp; }
```
Comment (Taylorr82) [inline ex2_hal/athena/equipment_handler/include/rtcmk.h:28]: Functions in headers shouldn't be declared static unless you mean to duplicate them in every file that includes them

**Review by Taylorr82 — COMMENTED** [review]
Is the static unix_timestamp initialized from the RTC on boot?

File: ex2_hal/athena/equipment_handler/include/rtcmk.h (hunk 1 above)
Comment (rcunrau) [reply ex2_hal/athena/equipment_handler/include/rtcmk.h:28]: Which I do because it's inline

**Comment by rcunrau** [comment]
> Is the static unix_timestamp initialized from the RTC on boot?

At least on my Athena, the RTC is initialized to 0 on boot (as is unix_timestamp). That surprised me, since I thought most RTCs had a battery so that they could retain the time across power outages etc., and I don't see that RTCMK_ResetTime is called anywhere. Or maybe we have ours wired so that it's always powered on the satellite? Anyway, I now zero the timestamp in ResetTime and added a call to RTCMK_GetUnix in rtc_daemon

**Comment by Taylorr82** [comment]
> > Is the static unix_timestamp initialized from the RTC on boot?
> 
> At least on my Athena, the RTC is initialized to 0 on boot (as is unix_timestamp). That surprised me, since I thought most RTCs had a battery so that they could retain the time across power outages etc., and I don't see that RTCMK_ResetTime is called anywhere. Or maybe we have ours wired so that it's always powered on the satellite? Anyway, I now zero the timestamp in ResetTime and added a call to RTCMK_GetUnix in rtc_daemon

On the satellite the RTC will only ever zero if there's a power loss event, so we're hoping for never. It is still possible to maintain RTC during a short power loss because of the giant capacitor on board

File: include/FreeRTOSConfig.h [hunk 2]
```diff
@@ -181,6 +181,10 @@ void vAssertCalled(unsigned long ulLine, const char * pcFile);
 extern BaseType_t prvRaisePrivilege( void );
 #define RAISE_PRIVILEGE BaseType_t xRunningPrivileged = prvRaisePrivilege ()
 #define RESET_PRIVILEGE if( xRunningPrivileged == 0 ) portSWITCH_TO_USER_MODE()
+
+void initializeProfiler();
```
Comment (Taylorr82) [inline include/FreeRTOSConfig.h:185]: These should be declared in a USER_CODE section instead. They will get overwritten if someone runs halcogen in this code

Comment (rcunrau) [reply include/FreeRTOSConfig.h:185]: Of course it doesn't show in a a context diff, but the prototypes are in "USER CODE BEGIN (5)"

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**